### PR TITLE
V7: event refactor

### PR DESCRIPTION
**Pull request by bengourley**
Mainly this PR updates `Event` to support multiple errors under the `event.errors` property.

It also has a general tidy up of the event's constructor, properties and types.

### CODE REVIEW

File: packages/core/event.js [hunk 1]
```diff
@@ -1,41 +1,49 @@
...
     this._user = {}
-    this.session = undefined
-    this.originalError = originalError
+    this._session = undefined
+
+    this.errors = [
+      {
+        errorClass: ensureString(errorClass),
+        errorMessage: ensureString(errorMessage),
+        type: jsRuntime,
```
Comment (djskinner) [inline packages/core/event.js]: Should this be defined on `interface Error` or is it an internal thing?

Just wondering if a plugin wanted to add a second error to this event would it need to remember to add  that `type: jsRuntime` or can we avoid that complexity by moving it to the `exceptions` projection in `toJSON`?

File: packages/core/types/event.d.ts [hunk 2]
```diff
@@ -59,4 +56,48 @@ interface Stackframe {
...
+  manufacturer?: string;
+  model?: string;
+  modelNumber?: string;
+  orientation?: string;
+  osName?: string;
+  osVersion?: string;
+  runtimeVersions: {
+    [key: string]: any;
+  };
+  time?: string;
```
[the remark this answers is not in the document]
Comment (bengourley) [reply packages/core/types/event.d.ts]: yeah it should be an ISO8601 string – looks like that detail got lost from the spec. I'll update it.

File: packages/core/types/event.d.ts [hunk 3]
```diff
@@ -59,4 +56,48 @@ interface Stackframe {
...
+interface Error {
+  errorClass: string;
+  errorMessage: string;
+  stacktrace: Stackframe[];
+}
+
+interface Device {
+  id?: string;
+  hostname?: string;
+  locale?: string;
```
[the remark this answers is not in the document]
Comment (bengourley) [reply packages/core/types/event.d.ts]: I think this is a typo in the spec, I'll update it

File: packages/core/event.js (hunk 1 above)
Comment (bengourley) [reply packages/core/event.js]: It's public and should be on the `Error` type so I'm adding it. In theory we can have a js error which is "caused by" a native error, and both may be in the array.

**Pull request merged**